src/www/containers/userprofile/UserProfile.js

```diff
@@ -0,0 +1,339 @@
+    categories: [{ name: "ABC" }, { name: "BCD" }],
+    photo: "/img/profilePhoto.jpg",
+  },
+  Action4: {
+    name: "ABCDE",
+    description:
+      "Description Description Description Description Description Description",
+    categories: [{ name: "ABC" }, { name: "BCD" }],
+    photo: "/img/profilePhoto.jpg",
+  },
+};
+
+const teams = {
+  Team1: {
+    name: "ABCDE",
+    description: "Description Description Description Description Description",
+    categories: [{ name: "ABC" }, { name: "BCD" }],
+    photo: "/img/teams.jpg",
+  },
+  Team2: {
+    name: "ABCDE",
+    description: "Description Description Description Description Description",
+    categories: [{ name: "ABC" }, { name: "BCD" }],
+    photo: "/img/teams.jpg",
+  },
+};
+
+const categories = {
+  Category1: {
+    name: "Hey",
+    photo: "/img/categories.jpg",
+  },
+  Category2: {
+    name: "Hey",
+    photo: "/img/categories.jpg",
+  },
+  Category3: {
+    name: "Hey",
+    photo: "/img/categories.jpg",
+  },
+};
+
+const mapState = (state) => ({
+  user: state.auth.user,
+});
+
+export default connect(
+  mapState,
+  undefined
+)(
+  withTranslation()(
+    class UserProfile extends Component {
+      state = {
+        activeTab: "actions",
+        attendedShown: 3,
+        editMode: "off",
+        organizedShown: 3,
+        teamsShown: 3,
+      };
+
+      showCards = (cards, count) => {
+        const { t } = this.props;
+        return Object.keys(cards).map((key, index) => {
+          const card = cards[key];
+          if (index >= count) {
+            return null;
+          }
+          return (
+            <li>
+              <span className="Card">
+                <img className="Card-image" src={card.photo} alt="" />
+                <div className="Card-body">
+                  <div>
+                    <h4>{card.name}</h4>
+                    <p>{card.description}</p>
+                  </div>
+                  <div className="Card-categories">
+                    <h4>{t("profile.categories")}:</h4>
+                    <ul>
+                      {Object.keys(card.categories).map((cKey) => {
+                        const category = card.categories[cKey];
+                        return <li>{category.name}</li>;
+                      })}
+                    </ul>
+                  </div>
+                </div>
+              </span>
+            </li>
+          );
+        });
+      };
+
+      expandActions = (actions, isAttended, minimum) => {
+        const { t } = this.props;
+        const count = isAttended
+          ? this.state.attendedShown
+          : this.state.organizedShown;
+        const list = isAttended ? "attendedShown" : "organizedShown";
+
+        if (Object.keys(actions).length < count) {
+          return null;
+        } else if (count <= minimum) {
+          return (
+            <p
+              onClick={() =>
+                this.setState({ [list]: Object.keys(actions).length })
```

Comment:
For some reason, when I delete the brackets the setState doesn't work as expected.

Reply:
Nevermind, it's probably needed since otherwise it's trying to set the field named "list" instead.
https://medium.com/@bretdoucette/understanding-this-setstate-name-value-a5ef7b4ea2b4